pom.xml

```diff
@@ -21,37 +21,43 @@
     <scala.version.prefix>2.10</scala.version.prefix>
     <hadoop.version>2.7.0</hadoop.version>
     <spark.version>1.6.1</spark.version>
+    <spark.hadoop.scope>provided</spark.hadoop.scope>
   </properties>
 
   <build>
+    <!--
+      The Guacamole JAR output by maven-jar-plugin, at the path configured here, can't run on its own; it needs some
+      dependency-classes shaded in, as configured elsewhere in this file. However, the shade plugin requires it to be
+      present in order to work.
+
+      As such, we write it to a non-default path here, and treat the minimum-viable Guacamole+shaded-deps JAR (built by
+      the default "guac" profile below) as the primary deliverable.
+     -->
+    <finalName>guacamole-plain-${project.version}</finalName>
     <plugins>
       <plugin>
         <groupId>org.scalatest</groupId>
         <artifactId>scalatest-maven-plugin</artifactId>
         <configuration>
-          <argLine>-Xmx4g -XX:MaxPermSize=256m</argLine>
+          <!-- Bump the memory used by scalatest. -->
+          <argLine>-Xmx4g</argLine>
         </configuration>
       </plugin>
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-shade-plugin</artifactId>
         <version>2.4.3</version>
         <configuration>
+
           <transformers>
+            <!-- Set "main" class. -->
             <transformer implementation="org.apache.maven.plugins.shade.resource.ManifestResourceTransformer">
               <mainClass>org.hammerlab.guacamole.Main</mainClass>
             </transformer>
-
-            <!-- Akka (which Spark depends on) needs this reference.conf munging in order to work correctly in -->
```

Comment:
yea we only need this in JARs that shade in spark/hadoop, so I put it in the `local` profile since they're `provided` otherwise